### PR TITLE
Add less_inheritance test for java, js, go

### PR DESCRIPTION
**Pull request by ulziibay**
WIth this our feature by lang matrix is almost non-empty.

<table style="text-align:center">
<tr>
<td>

</td>
<td>
<b>go</b>
</td>
<td>
<b>python</b>
</td>
<td>
<b>js</b>
</td>
<td>
<b>java</b>
</td>
<td>
<b>php</b>
</td>
<td>
<b>ml</b>
</td>
<td>
<b>cpp</b>
</td>
<td>
<b>c</b>
</td>
</tr>
<tr>
<td>
dots
</td>
<td>
6
</td>
<td>
9
</td>
<td>
6
</td>
<td>
5
</td>
<td>
0
</td>
<td>
0
</td>
<td>
0
</td>
<td>
3
</td>
</tr>
<tr>
<td>
equivalence
</td>
<td>
2
</td>
<td>
3
</td>
<td>
1
</td>
<td>
1
</td>
<td>
0
</td>
<td>
0
</td>
<td>
0
</td>
<td>
1
</td>
</tr>
<tr>
<td>
metavar
</td>
<td>
5
</td>
<td>
8
</td>
<td>
7
</td>
<td>
7
</td>
<td>
0
</td>
<td>
0
</td>
<td>
0
</td>
<td>
7
</td>
</tr>
<tr>
<td>
misc
</td>
<td>
1
</td>
<td>
11
</td>
<td>
1
</td>
<td>
1
</td>
<td>
0
</td>
<td>
0
</td>
<td>
0
</td>
<td>
0
</td>
</tr>
<tr>
<td>
less
</td>
<td>
2
</td>
<td>
2
</td>
<td>
1
</td>
<td>
1
</td>
<td>
0
</td>
<td>
0
</td>
<td>
0
</td>
<td>
0
</td>
</tr>
</table>

### CODE REVIEW

File: tests/generate.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# /usr/bin/env python3
```
Comment (aryx) [inline tests/generate.py]: maybe rename generate_test_matrix.py

File: tests/go/less_atts.sgrep [hunk 2]
```diff
@@ -0,0 +1,3 @@
+func $F(...) {
```
Comment (aryx) [inline tests/go/less_atts.sgrep]: what does atts mean? And why did you prefix this test with a less? It looks more like a dots_xxx.
The spirit of the less_xxx is to say that a pattern can have less details that the concrete file.
For example writing the pattern class A { ...} will also match code where the code has the extra detail
about inheritnace like class A extends B {}. We do not force the user to write class A extends $X { ...} .

File: tests/go/less_inherits.TODO [hunk 3]
```diff
@@ -0,0 +1,3 @@
+type struct A {
```
Comment (aryx) [inline tests/go/less_inherits.TODO]: Go does not really have inheritance like the other languages. It does not fit in the less_xxx category.
However having dots for fields is a nice test to have, so maybe rename to
dots_struct_fields.TODO

File: tests/java/less_inheritance.java.TODO [hunk 4]
```diff
@@ -0,0 +1,19 @@
+// ERROR: match
+public class A {
+  public static foo() {
+    return 1;
+  }
+}
+
+// ERROR: match
+public class B extends A {
```
Comment (aryx) [inline tests/java/less_inheritance.java.TODO]: No, this should be class A extends B { ...}
we do not want class A { .. } to match public class B extends A { }

File: tests/js/less_inheritance.js.TODO [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    this.name = name;
+  }
+
+  foo() {
+    return 'foo';
+  }
+}
+
+//ERROR:
+class B extends A {
```
Comment (aryx) [inline tests/js/less_inheritance.js.TODO]: no, should be class A extends B.

File: tests/go/less_atts.sgrep (hunk 2 above)
Comment (ulziibay) [reply tests/go/less_atts.sgrep]: Move to tests/go/dots_func_attributes.sgrep to clarify!

File: tests/java/less_inheritance.java.TODO (hunk 4 above)
Comment (ulziibay) [reply tests/java/less_inheritance.java.TODO]: ok

File: tests/js/less_inheritance.js.TODO (hunk 5 above)
Comment (ulziibay) [reply tests/js/less_inheritance.js.TODO]: addressed

File: tests/go/less_inherits.TODO (hunk 3 above)
Comment (ulziibay) [reply tests/go/less_inherits.TODO]: Ok addressed

Comment (ulziibay) [reply tests/go/less_inherits.TODO]: Ok addressed

File: tests/go/dots_attributes.sgrep [hunk 6]
```diff
@@ -0,0 +1,3 @@
+func $F(...) {
```
Comment (aryx) [inline tests/go/dots_attributes.sgrep]: still don't understand what it has to do with attributes (a.k.a annotations).

**Comment by ulziibay** [comment]
Linking #239

**Pull request merged**